onboard/nav/avoidances.cpp

```diff
@@ -0,0 +1,164 @@
+/* Searches.cpp */
+
+#include "avoidances.hpp"
+#include "utilities.hpp"
+#include "stateMachine.hpp"
+
+#include <iostream>
+#include <cmath>
+#include <stack>
+
+/*************************************************************************/
+/* Avoidance Factory */
+/*************************************************************************/
+// Avoidance Factory to allow for creation of obstacle avoidance objects.
+// Allows for easy change from obstacle avoidance algorithm to another
+Avoidance* AvoidFactory( StateMachine* stateMachine, AvoidanceType type )
+{
+    Avoidance* avoid = nullptr;
+    switch (type)
+    {
+        case AvoidanceType::ORIGINAL:
+            avoid = new Original( stateMachine );
+            break;
+
+        case AvoidanceType::UNKNOWN:
+            std::cerr << "Unkown Search Type. Defaulting to original\n";
+            avoid = new Original( stateMachine );
+            break;
+    }
+    return avoid;
+}
+
+/*************************************************************************/
+/* Original Obstacle Avoidance Algorithm
```

Comment:
Again I suggest using a better name than `Original` in favor of something more descriptive. If nothing strikes you my suggestion is `Simple` since we are performing the most simple of avoidance algorithms we can by inserting one point into our path to avoid create a new path to avoid the obstacle. I think the comment should also include something similar to this so as to describe how this obstacle avoidance algorithm solves our problem.